buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeMultiplatformExtensionImpl.kt

```diff
@@ -121,3 +131,93 @@ open class AndroidXComposeMultiplatformExtensionImpl @Inject constructor(
+
+
+fun OELPublishingPrototype(project: Project) {
+    if (!project.experimentalOELPublication()) return
+
+    if (project.experimentalOELPublication() && (project.oelAndroidxVersion() == null)) {
+        error("androidx version should be specified for OEL publications")
+    }
+
+
+    val ext = project.multiplatformExtension ?: error("expected a multiplatform project")
+
+    ext.targets.all { target ->
+        if (target is KotlinAndroidTarget) {
+            project.publishAndroidxReference(target)
+        }
+    }
+}
+
+// FIXME: reflection access! Some API in Kotlin is needed
+@Suppress("unchecked_cast")
+private val KotlinTarget.kotlinComponents: Iterable<KotlinTargetComponent>
+    get() = javaClass.kotlin.memberProperties
+        .single { it.name == "kotlinComponents" }
+        .get(this) as Iterable<KotlinTargetComponent>
+
+
+@Suppress("unchecked_cast")
+private fun Project.publishAndroidxReference(target: KotlinTarget) {
+    afterEvaluate {
+        target.kotlinComponents.forEach { component ->
+            val componentName = component.name
+
+            val multiplatformExtension =
+                extensions.findByType(KotlinMultiplatformExtension::class.java)
+                    ?: error("Expected a multiplatform project")
+
+            if (component is KotlinVariant)
+                component.publishable = false
+
+            val usages = when (component) {
+                is KotlinVariant -> component.usages
+                is JointAndroidKotlinTargetComponent -> component.usages
+                else -> emptyList()
+            }
+
+            extensions.getByType(PublishingExtension::class.java)
+                .publications.withType(DefaultMavenPublication::class.java)
+                // isAlias is needed for Gradle to ignore the fact that there's a
+                // publication that is not referenced as an available-at variant of the root module
+                // and has the Maven coordinates that are different from those of the root module
+                // FIXME: internal Gradle API! We would rather not create the publications,
+                //        but some API for that is needed in the Kotlin Gradle plugin
+                .all { publication ->
+                    if (publication.name == componentName) {
+                        publication.isAlias = true
+                    }
+                }
+
+            usages.forEach {    usage ->
+                val configurationName = usage.name + "-published"
+
+                configurations.matching{it.name == configurationName}.all() { conf ->
+                    conf.artifacts.clear()
+                    conf.dependencies.clear()
+                    conf.setExtendsFrom(emptyList())
+                    var version = if (target.project.group.toString().contains("org.jetbrains.compose.material3")) "1.0.0-alpha01" else target.project.oelAndroidxVersion()!!
```

Comment:
Please, add changes from this [commit](https://github.com/JetBrains/androidx/commit/d770f4e2d877a9b9c907a353ac93e7239e7ac702)

Reply:
done